src/components/submission/diff-item.vue

```diff
@@ -0,0 +1,267 @@
+      const basicPath = `/${fullPathStr}`;
+      if (this.binaryFieldPaths.has(basicPath))
+        return true;
+      return false;
+    }
+  },
+  methods: {
+    flattenDiff(node, typeOfChange, prefix = []) {
+      // Flattens the inner diff (JSON of an entire added/removed tree structure)
+      // by recursively traversing the tree
+      const changes = [];
+      // eslint-disable-next-line guard-for-in
+      for (const k in node) {
+        const path = prefix.concat(k);
+        if (Array.isArray(node)) {
+          path.pop(); // pop off the index (e.g. 1 of toy #1)
+          path.pop(); // pop off the name (e.g. toy of toy #1)
+          path.push([prefix[prefix.length - 1], Number.parseInt(k, 10)]);
+        }
+        if (typeof (node[k]) === 'object') {
+          changes.push(...this.flattenDiff(node[k], typeOfChange, path));
+        } else if (typeOfChange === 'added') {
+          changes.push({ path, old: null, new: node[k] });
+        } else {
+          changes.push({ path, old: node[k], new: null });
+        }
+      }
+      return changes;
+    },
+    binaryHref(value) {
+      return apiPaths.submissionAttachment(
+        this.projectId,
+        this.xmlFormId,
+        false,
+        this.instanceId,
+        value
+      );
+    },
+    formatData(content) {
+      const length = 240;
+      return content.length > length ? content.slice(0, length).concat('...') : content;
+    }
+  }
+};
+</script>
+
+<style lang="scss">
+@import '../../assets/scss/mixins';
+
+.submission-diff-item {
+
+  &.outer-item {
+    padding: 5px;
+    border-bottom: 1px solid #ccc;
+
+    &:last-child { border-bottom: 0; }
+  }
+
+  &.inner-item {
+    padding: 0px;
+    border-bottom: 1px solid #ddd;
+
+    &:last-child { border-bottom: 0; }
+  }
+
+  .full-path, .field-name {
+    font-family: $font-family-monospace;
+  }
+
+  .full-path {
+    font-size: 12px;
+    color: #333;
+    margin: 10px 10px -5px 10px;
+  }
+
+  .field-separator, .field-counter{
+    color: #999;
+  }
+
+  .diff-details {
+    display: flex;
+    font-size: 15px;
+  }
+
+  .field-name, .nested-change-type {
+    width: 150px;
+    margin: 10px;
+    font-size: 14px;
+    flex-shrink: 0;
+  }
+
+  .field-name {
+    color: #000;
+    @include text-overflow-ellipsis;
+  }
+
+  .old-to-new { margin: 10px; }
+  .icon-arrow-circle-right { color: #888; padding: 10px; }
+
+  .data-old { color: $color-danger-dark; }
+  .data-new { color: $color-success-dark; }
+
+  .deleted { color: $color-danger-dark; }
+  .added { color: $color-success-dark; }
+
+  .data-empty {
+    background-color: #bbb;
+    color: white;
+    padding: 5px;
+    border-radius: 2px;
+  }
+}
+
+</style>
+
+<i18n lang="json5">
+{
+  "en": {
+    // The description of how a specific field in a form submission changed
+    "editCaption": {
+      "added": "(added)",
+      "deleted": "(deleted)",
+    },
+    "empty": "empty", // Text showing that a value in a submission edit is empty
```

Comment:
The i18n comment mechanism is pretty particular: the comment needs to be before the property name in order to be picked up. (You can see that the comment doesn't currently appear in `strings_en.json`.)